### ISSUE TITLE
Question mark unrendered characters in "assistance listings" on Opp Listing page

### ISSUE THREAD

**Issue by lucasmbrown-usds** [HHS/simpler-grants-gov#2591]
### Summary

Some of the characters in the "assistance listing" field are being rendered as question marks. Is this a problem with the source data or our rendering? 

![Image](https://github.com/user-attachments/assets/6d9c1b33-3ea5-4075-a3e2-568a4ad89de0)

### Reproducibility

I tried multiple times and saw the bug every time

### How to reproduce the bug

Visit https://simpler.grants.gov/opportunity/356595. 

### Browser

Chrome

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by mdragon** [comment]
Just noting this displays the same for me on the existing grants.gov page so it might be a data issue not a Simpler site issue ![Image](https://github.com/user-attachments/assets/5a168a71-ea8c-4731-a349-bbfc9780be3e)
https://simpler.grants.gov/opportunity/356595

**Comment by mdragon** [comment]
![Image](https://github.com/user-attachments/assets/14d8ced4-58c0-4b86-9ec1-811309622ff3)

These extended characters are present in our operational database for this record. The fact that the existing site also renders them seems to indicate there was an encoding issue with the creation of this data on the Oracle side, and we're just pulling that through to our side.

**Comment by mxk0** [comment]
@mdragon should we close this out?

**Comment by mdragon** [comment]
@chouinar would you agree with my hypothesis here? This is unlikely to indicate an issue with our data movement setup and is likely not fixable on Simpler's side?

**Comment by chouinar** [comment]
Right, I believe that character is the equivalent of a fallback if you try to put something in that isn't supported in your encoding.

Going even further into the source data, I downloaded the [csv from sam.gov](https://sam.gov/data-services/Assistance%20Listings/grantsgov/2024/12-Dec?privacy=Public) (assistance listings come from there) and see "Geographic Programs ñ Gulf of Mexico Program" - guessing that special character got turned into three of the unknown characters at some point.

**Comment by chouinar** [comment]
It renders fine on the browser in Sam.gov: https://sam.gov/fal/369a25a7bdfe48658a65172e374bd934/view

So, the issue is likely in sam.gov's export process if anything

**Comment by chouinar** [comment]
As a side note, the assistance listing data is setup a bit oddly in our system.

There are two relevant tables in the legacy system, one for all the assistance listings for an opportunity, and another for the raw assistance listing info itself. The names of each of the assistance listings is duplicated onto the opportunity-assistance listing data, when in reality that table should just be a link to the assistance listing table (would be good to ask why the legacy system has that approach).

If we had an assistance listing table, we could just fix the text on our side, so might be worth looking into that angle.

**Comment by mdragon** [comment]
@lucasmbrown-usds I'm thinking given the root cause of this that we close out this ticket and maybe turn Michael's last comment here into a backlog ticket around supporting Simpler "overrides" of upstream data to deal with typos or stuff like this that might be hard to get fixed up stream but does impact UX in the Grants ecosystem?

**Comment by lucasmbrown-usds** [comment]
Created a ticket with Sam.gov help desk: INC-GSAFSD11490181 

Ticket content is: 

> As you can see from our discussion at https://github.com/HHS/simpler-grants-gov/issues/2591, there are some problems in the data exports from Sam.gov. 
> 
> This entity renders a dash in the name of the entity: https://sam.gov/fal/369a25a7bdfe48658a65172e374bd934/view. It says "Geographic Programs – Gulf of Mexico Program".
> 
> We downloaded the csv from sam.gov (assistance listings come from there) and see this rendered "Geographic Programs ñ Gulf of Mexico Program". 
> 
> Could you assist in tracking cleaning up the data so that dashes get rendered properly? It is ruining the rendering on our site: https://simpler.grants.gov/opportunity/356595 
> 
> Thank you!

**Comment by lucasmbrown-usds** [comment]
I have been communicating with the Sam.gov help desk

**Comment by chouinar** [comment]
An idea - I'm going to look through the data to see if these always end up as the `/ufffd` character. If so, during the transformation process, we can just make it so those get replaced with a simple `-`. We'd also need to cleanup the data in our DB, but I'll leave that for a follow-up.

**Comment by chouinar** [comment]
In prod, I see 81 results searching for `select * from api.opportunity_assistance_listing where  program_title like '%�%' order by created_at desc`

A few of the assistance listings I see:
https://sam.gov/fal/369a25a7bdfe48658a65172e374bd934/view
https://sam.gov/fal/3b0519bf9aeb43779df9d6e8d46c6457/view
https://sam.gov/fal/c7b2285061aa42b68d6ac3157cef7cfd/view - in this one the quotes around WANTO are what ends up wrong (so two sets of 3)
https://sam.gov/fal/bdab05af045b42b192f69fbfe1a7e346/view - in this one the issue is the apostrophe
https://sam.gov/fal/a4f2ef594f27431baa2eed0adf9d2750/view - also the apostrophe
https://sam.gov/fal/d917cee4e60543f68ac1fda7da0bb261/view - also an apostrophe

So it seems like it's a mix of dash, apostrophe and quotes that cause the issue. Don't think we can replace these characters with anything in that case, if we wanted to do something on our side, the best we could do is just getting rid of the character entirely when transforming.

Do we want to go with that approach?

**Comment by chouinar** [comment]
The full results of everything with the weird characters:
```
10.210 | Higher Education ��� Graduate Fellowships Grant Program
10.244 | Special Supplemental Nutrition Program for Women, Infants, and Children ��� National Workforce Strategy Development
10.545 | Farmers��� Market Supplemental Nutrition Assistance Program Support Grants
11.038 | Public Wireless Supply Chain Innovation Fund Grant Program ��� Expanding Testing and Evaluation
15.507 | WaterSMART (Sustain and Manage America���s Resources for Tomorrow)
15.560 | SECURE Water Act ��� Research Agreements
15.945 | Cooperative Research and Training Programs ��� Resources of the National Park System
17.701 | Women in Apprenticeship and Nontraditional Occupations (���WANTO���) Technical Assistance Grant Program
17.805 | Homeless Veterans��� Reintegration Program
19.441 | ECA ��� American Spaces
20.115 | Fueling Aviation���s Sustainable Transition via Sustainable Aviation Fuels (FAST-SAF) and Low-Emissions Aviation Technologies (FAST-Tech) Grant
66.448 | Drinking Water System Infrastructure Resilience and Sustainability Program ��� SDWA 1459A(l)
66.475 | Geographic Programs ��� Gulf of Mexico Program
84.428 | Augustus F. Hawkins Centers of Excellence���Teacher Preparation and Development
93.346 | Enhancing the Logical Observation Identifiers Names and Codes (LOINC��) Standard to meet U.S. Interoperability Needs
93.421 | Strengthening Public Health Systems and Services through National Partnerships to Improve and Protect the Nation���s Health
93.431 | Networking2Save���: CDC���s National Network Approach to Preventing and Controlling Tobacco-related Cancers in Special Populations
93.470 | Alzheimer���s Disease Program Initiative (ADPI)
93.686 | Ending the HIV Epidemic: A Plan for America ��� Ryan White HIV/AIDS Program Parts A and B
93.734 | Empowering Older Adults and Adults with Disabilities through Chronic Disease Self-Management Education Programs ��� financed by Prevention and Public Health Funds (PPHF)
```

**Comment by mxk0** [comment]
This issue is happening somewhere between the data source and Grants.gov. We could manually fix these 20 records with a transformation on our own copy of the data, but that isn't a sustainable fix. Given the low number of impacted records and the relatively minor UX impact, we don't think this is worth acting on.

**Comment by mdragon** [comment]
Just to flag, another alternative to fix or not fix is we could start parsing this SAM data directly, rather than pulling it through from the existing system. Then any issue with the data is getting handled at the root of that parsing in Simpler and we still "solve" the issue but in a more forward looking way.

**Comment by chouinar** [comment]
@mdragon - We'll need to do that at some point, but it will require a few changes to the data structure.

Right now we get the program name from the `opportunity_assistance_listing` table which _should_ just be a many-to-many mapping between opportunity and assistance listings, but we don't have a cache of assistance listings, we just have the data for each of them copied to this table.

If we had an `assistance_listing` table that was just a daily cache of the sam.gov data, we'd also need to adjust the schema of the `opportunity_assistance_listing` table to actually be a many-to-many mapping. In the long-term we 100% should do this - might be worthwhile to make the tickets now even if we don't deal with it right this second.

(On the note of odd mappings, we should also fix the opportunity connection to the agency table to use IDs rather than agency code).
